packages/ui/lib/utils.ts

```diff
@@ -21,3 +23,9 @@ export function humanFileSize(bytes: number, si = false, dp = 1) {
 
   return bytes.toFixed(dp) + ' ' + units[u];
 }
+
+export function usePluginProvider(): PluginProvider {
+  const pluginProvider = new PluginProvider();
```

Comment:
This will cause a new PluginProvider to be created each time this is called, I think you want to do is create a singleton which you need to use the custom hooks pattern from reach instead. https://reactjs.org/docs/hooks-custom.html#extracting-a-custom-hook

Reply:
updated this, let me know what you tihnk